chat-client/src/client/chat.test.ts

```diff
@@ -0,0 +1,69 @@
+import { injectJSDOM } from '../test/jsDomInjector'
+// This needs to be run before all other imports so that mynah ui gets loaded inside of jsdom
+injectJSDOM()
+
+import { afterEach } from 'mocha'
+import { assert } from 'sinon'
+import { NEW_TAB_CREATED, TAB_CHANGED, TAB_REMOVED, UI_IS_READY } from '../contracts/serverContracts'
+import { SEND_TO_PROMPT, TAB_ID_RECEIVED } from '../contracts/uiContracts'
+import { createChat } from './chat'
+import sinon = require('sinon')
+
+describe('Chat', () => {
+    const sandbox = sinon.createSandbox()
+    let clientApi: { postMessage: any }
+
+    beforeEach(() => {
+        clientApi = {
+            postMessage: sandbox.stub(),
+        }
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    it('chat initialized publishes ready event', () => {
+        createChat(clientApi)
```

Comment:
What does `createChat` do exactly? Create the Chat UI? The Chat JSON/RPC connection? The Chat Proxy/Router/Client? Can we make this more descriptive of which component it is?

Reply:
It injects chat UI with `postMessage` listener and the logic around it into the destination. This is also the function that is exposed to destination - I did not want it to be too specific intentionally. Will `createChatClient` look better?